src/librustc_typeck/check/mod.rs

```diff
@@ -3819,6 +3819,41 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         pointing_at_return_type
     }
 
+    fn suggest_fn_call(
+        &self,
+        err: &mut DiagnosticBuilder<'tcx>,
+        expr: &hir::Expr,
+        expected: Ty<'tcx>,
+        found: Ty<'tcx>,
+    ) -> bool {
```

Comment:
The body could use some commentary in various place and a doc comment on `suggest_fn_call` showing a sample diagnostic.